components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/AuthorizationCodeDAOImpl.java

```diff
@@ -83,16 +84,32 @@ public void insertAuthorizationCode(String authzCode, String consumerKey, String
         Connection connection = IdentityDatabaseUtil.getDBConnection();
         PreparedStatement prepStmt = null;
         String userDomain = authzCodeDO.getAuthorizedUser().getUserStoreDomain();
-        String authenticatedIDP = authzCodeDO.getAuthorizedUser().getFederatedIdPName();
-
-        if (!OAuthServerConfiguration.getInstance().isMapFederatedUsersToLocal() && authzCodeDO.getAuthorizedUser()
-                .isFederatedUser()) {
-            userDomain = OAuth2Util.getFederatedUserDomain(authenticatedIDP);
+        String authenticatedIDP;
+        if (OAuth2ServiceComponentHolder.isIDPIdColumnEnabled()) {
```

Comment:
Move getting authenticatedIDP logic to a private method

Reply:
@mefarazath done please check private methods at bottom